src/libvirtApi/storageVolume.js

```diff
@@ -55,62 +52,36 @@ export function storageVolumeCreateAndAttach({
-    return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'StoragePoolLookupByName', [poolName], { timeout, type: 's' })
-            .then(storagePoolPath => {
-                return call(connectionName, storagePoolPath[0], 'org.libvirt.StoragePool', 'ListStorageVolumes', [0], { timeout, type: 'u' });
-            })
-            .then((objPaths) => {
-                const volumes = [];
-                const storageVolumesPropsPromises = [];
-
-                for (let i = 0; i < objPaths[0].length; i++) {
-                    const objPath = objPaths[0][i];
-
-                    storageVolumesPropsPromises.push(
-                        call(connectionName, objPath, 'org.libvirt.StorageVol', 'GetXMLDesc', [0], { timeout, type: 'u' })
-                    );
-                }
-
-                // WA to avoid Promise.all() fail-fast behavior
-                const toResultObject = (promise) => {
-                    return promise
-                            .then(result => ({ success: true, result }))
-                            .catch(error => ({ success: false, error }));
-                };
+export async function storageVolumeGetAll({ connectionName, poolName }) {
+    try {
+        const [storagePoolPath] = await call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'StoragePoolLookupByName',
+                                             [poolName], { timeout, type: 's' });
+        const [objPaths] = await call(connectionName, storagePoolPath, 'org.libvirt.StoragePool', 'ListStorageVolumes', [0],
+                                      { timeout, type: 'u' });
 
-                return Promise.all(storageVolumesPropsPromises.map(toResultObject)).then(volumeXmlList => {
-                    for (let i = 0; i < volumeXmlList.length; i++) {
-                        if (volumeXmlList[i].success) {
-                            const volumeXml = volumeXmlList[i].result[0];
-                            const dumpxmlParams = parseStorageVolumeDumpxml(connectionName, volumeXml);
+        const storageVolumesPropsPromises = objPaths.map(objPath =>
+            call(connectionName, objPath, 'org.libvirt.StorageVol', 'GetXMLDesc', [0], { timeout, type: 'u' })
+        );
 
-                            volumes.push(dumpxmlParams);
-                        }
-                    }
-                    return Promise.resolve(volumes);
-                });
-            })
-            .catch(ex => console.warn("GET_STORAGE_VOLUMES action failed for pool", poolName, ":", ex.toString()));
+        const volumeXmlList = await Promise.allSettled(storageVolumesPropsPromises);
+        return volumeXmlList
+                .filter(vol => vol.status === 'fulfilled')
+                .map(vol => parseStorageVolumeDumpxml(connectionName, vol.value[0]));
+    } catch (ex) {
+        console.warn("GET_STORAGE_VOLUMES action failed for pool", poolName, ":", ex.toString());
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1280-20240125-124532-d8538f72-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)